### PR TITLE
rename `ShaderModel` enums

### PR DESCRIPTION
**Pull request by pixelflinger**
GL_ES_30   becomes MOBILE
GL_CORE_41 becomes DESKTOP

ShaderModel controls which flavor of GLSL (GL or ES) is used both when 
reading and outputting materials.

It's also used to set default quality settings and the default precision
of all fragment shaders.

Technically, Vulkan and Metal don't need this distinction, but the GL
backend does.

### CODE REVIEW

**Review by prideout — CHANGES_REQUESTED** [review]
We cast this enum to an int in `Material::Builder::build` and compare the result against a field that was extracted from the material package.  This implies that we need a material version bump, right?  Alternatively, maybe we could just start the enum at 1 instead of 0?

File: filament/backend/src/metal/MetalDriver.mm [hunk 1]
```diff
@@ -552,7 +552,7 @@
 #if defined(IOS)
     return ShaderModel::GL_ES_30;
```
Comment (bejado) [inline filament/backend/src/metal/MetalDriver.mm:553]: Please update this to `ShaderModel::MOBILE`.

File: filament/backend/src/noop/NoopDriver.cpp [hunk 2]
```diff
@@ -35,7 +35,7 @@ ShaderModel NoopDriver::getShaderModel() const noexcept {
 #if defined(__ANDROID__) || defined(IOS) || defined(__EMSCRIPTEN__)
     return ShaderModel::GL_ES_30;
```
Comment (bejado) [inline filament/backend/src/noop/NoopDriver.cpp:36]: Please update this to `ShaderModel::MOBILE`.

**Review by bejado — CHANGES_REQUESTED** [review]

File: filament/backend/src/metal/MetalDriver.mm (hunk 1 above)
Comment (pixelflinger) [reply filament/backend/src/metal/MetalDriver.mm]: done

File: filament/backend/src/noop/NoopDriver.cpp (hunk 2 above)
Comment (pixelflinger) [reply filament/backend/src/noop/NoopDriver.cpp]: done

**Comment by pixelflinger** [comment]
@prideout @bejado very good catches! thanks. all fixed.

**Review by prideout — APPROVED** [review]

**Review by bejado — APPROVED** [review]

**Pull request merged**